lib/spack/spack/build_systems/python.py

```diff
@@ -88,6 +91,26 @@ def configure(self, spec, prefix):
 
     py_namespace = None
 
+    @property
+    def homepage(self):
+        if self.pypi:
+            name = self.pypi.split('/')[0]
+            return 'https://pypi.org/project/' + name + '/'
+
+    @property
+    def url(self):
```

Comment:
oh I love this. allows non-pypi packages to keep working just fine.

Reply:
Do we want to add:
```python
else:
    raise AttributeError('Must define `pypi` or `url`')
```
for these fields? We do this for other mixins like sourceforce/gnu mirrors. But I don't think it's strictly necessary.